haskell-src/Concordium/Types.hs

```diff
@@ -711,7 +711,23 @@ instance Show Address where
 
 -- | Time in seconds since the unix epoch
 newtype TransactionTime = TransactionTime {ttsSeconds :: Word64}
-    deriving (Show, Read, Eq, Num, Ord, FromJSON, ToJSON, Real, Enum, Integral) via Word64
+    deriving (Show, Read, Eq, Num, Ord, Real, Enum, Integral) via Word64
+
+-- Implement `ToJSON` instance for `TransactionTime`.
+instance ToJSON TransactionTime where
+    toJSON (TransactionTime seconds) =
+        String $ T.pack $ formatTime defaultTimeLocale "%FT%T%QZ+00:00" (posixSecondsToUTCTime (fromIntegral seconds))
```

Comment:
**Suggestion:** Avoid formatting dates manually, this is not the first time we depend on this format, so I expect we have something already.
Maybe this is good enough https://hackage.haskell.org/package/time-1.14/docs/Data-Time-Format-ISO8601.html